src/udev/udevd.c

```diff
@@ -265,6 +270,10 @@ static int worker_new(Worker **ret, Manager *manager, sd_device_monitor *worker_
         assert(worker_monitor);
         assert(pid > 1);
 
+        r = sd_event_add_child(manager->event, &s, pid, WEXITED, on_sigchld, manager);
```

Comment:
hmm, shouldn't this get the newly allocated worker rather than the manager as `userdata` argument?